models.js

```diff
@@ -64,12 +64,28 @@ exports.fetchArticleById = (id) => {
  
 }
 exports.insertingComments = (body, id) => {
-    return db.query(`
-    INSERT INTO comments (body,author,article_id)
-    VALUES ($1, $2, $3)
-    RETURNING *;
-   `, [body.body, body.author, id])
-   .then((result)=> {
-    return result.rows[0]
-   })
+    return checkArticleExists(id)
+    .then(()=>{
+
+        if (Object.keys(body).length === 0){
+            return Promise.reject({
+                status : 400,
+                msg: "Bad, bad Request"
+            })
+        }
+        if ("author" in body && "body" in body ){
+            return checkAuthorExists(body.author)
+            .then(()=>{
+               return db.query(`
+                INSERT INTO comments (body,author,article_id)
+                VALUES ($1, $2, $3)
+                RETURNING *;
+               `, [body.body, body.author, id])
+               .then((result)=> {
+                return result.rows[0]
+               })
+            })
+        }
+
+    })
 }
```

Comment:
what happens in the event that you dont go into that if statement?

Reply:
@MKD01 am I right thinking that I put an else statement where I reject the promise (indicating to the user that they must insert body AND author) will also resolve the other comment you have left?

Reply:
@MKD01 that test would try posting a comment with no body, just author